phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -3207,17 +3226,37 @@ public MetaDataMutationResult updateMutation(PTable table, byte[][] rowKeyMetaDa
                     } 
                     if (type == PTableType.VIEW
                             && EncodedColumnsUtil.usesEncodedColumnNames(table)) {
-                        /*
-                         * When adding a column to a view that uses encoded column name scheme, we
-                         * need to modify the CQ counters stored in the view's physical table. So to
-                         * make sure clients get the latest PTable, we need to invalidate the cache
-                         * entry.
-                         */
-                        invalidateList.add(new ImmutableBytesPtr(MetaDataUtil
-                                .getPhysicalTableRowForView(table)));
-
-
-
+                                /*
+                                 * When adding a column to a view that uses encoded column name
+                                 * scheme, we need to modify the CQ counters stored in the view's
+                                 * physical table. So to make sure clients get the latest PTable, we
+                                 * need to invalidate the cache entry.
+                                 */
+                                invalidateList.add(new ImmutableBytesPtr(
+                                        MetaDataUtil.getPhysicalTableRowForView(table)));
+                                byte[] parentSchemaName =
+                                        SchemaUtil.getSchemaNameFromFullName(
+                                            table.getPhysicalName().getString()).getBytes();
+                                byte[] parentTableName =
+                                        SchemaUtil
+                                                .getTableNameFromFullName(
+                                                    (table.getPhysicalName().getString()))
+                                                .getBytes();
+                                PTable parentTable =
+                                        doGetTable(null, parentSchemaName, parentTableName,
```

Comment:
doGetTable expects table key to be in the same region, so if a view is created in different schema than the parent schema, will it work in case of mult-region SYSTEM.CATALOG table.